env/ipp_aes_ctr_provider.cc

```diff
@@ -0,0 +1,131 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  Copyright (c) 2020 Intel Corporation
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#ifdef IPPCP
+
+#ifndef ROCKSDB_LITE
+
+#include "rocksdb/ipp_aes_ctr_provider.h"
+
+#include "util/random.h"
+
+#endif
+
+namespace ROCKSDB_NAMESPACE {
+
+#ifndef ROCKSDB_LITE
+
+IppAESCTRCipherStream::IppAESCTRCipherStream(IppsAESSpec* aes_ctx,
+                                             const char* init_vector)
+    : aes_ctx_(aes_ctx) {
+#if defined(HAVE_SSE42) || defined(HAVE_SSE2)
+  init_vector_ = _mm_loadu_si128((__m128i*)init_vector);
+#else
+  init_vector_ = init_vector;
+#endif
```

Comment:
From the code, it is not clear to me what the lifetime of "init_vector" is and who "owns/deletes" it.  Can you clarify that via some comments somewhere please?

Also, should this be "const IppsAESSpec* aes_ctx"?